File: 201-300/240/Solution.java
//可将矩阵看做二叉搜索树
class Solution {
    public boolean searchMatrix(int[][] matrix, int target) {
        int i = 0, j = matrix[0].length - 1;
        while(i < matrix.length && j >= 0){
            if(matrix[i][j] == target) return true;
            if(matrix[i][j] < target) i++;
            else j--;
        }
        return false;
    }
}

File: 201-300/240/solution.go
//可将矩阵看做二叉搜索树
func searchMatrix(matrix [][]int, target int) bool {
    i, j := 0, len(matrix[0])-1
    for i < len(matrix) && j >= 0 {
        if target == matrix[i][j] {return true}
        if target > matrix[i][j] {i++} else {j--}
    }
    return false
}
